solar/solar/core/transports/base.py

```diff
@@ -40,6 +40,44 @@ def run(self, transport):
             self._executor(transport)
 
 
+class SolarRunResultWrp(object):
+
+    def __init__(self, name):
+        self.name = name
+
+    def __get__(self, obj, objtype):
+        res = obj._result
+        if isinstance(res, dict):
+            try:
+                return res[self.name]
+            except KeyError:
+                # Let's keep the same exceptions
+                raise AttributeError(self.name)
+        return getattr(obj._result, self.name)
+
+
+class SolarRunResult(object):
+
+    def __init__(self, result):
+        self._result = result
+
+    failed = SolarRunResultWrp('failed')
+    stdout = SolarRunResultWrp('stdout')
+    stderr = SolarRunResultWrp('stderr')
+    succeeded = SolarRunResultWrp('succeeded')
+    command = SolarRunResultWrp('command')
+    real_command = SolarRunResultWrp('real_command')
+    return_code = SolarRunResultWrp('return_code')
+
+    def __str__(self):
+        if self.failed:
+            return str(self.stderr)
+        return str(self.stdout)
+
+    def split(self, *args, **kwargs):
```

Comment:
in fabric split also applies to stdout ONLY. That's why. I f you want to split stdout,stderr then you need to call it directly. (exactly like in fabric api)